src: add a method to get string encoding info (#56147)
nodejs/node: doc/api/v8.md

}, 1000);
```

## `v8.isStringOneByteRepresentation(content)`

<!-- YAML
added: REPLACEME
-->

* `content` {string}
* Returns: {boolean}

> V8 only supports `Latin-1/ISO-8859-1` and `UTF16` as the underlying representations.

Reviewer: ```suggestion
V8 only supports `Latin-1/ISO-8859-1` and `UTF16` as the underlying representation of a string.
```

  cursor;

  constructor() {
    this.buffer = Buffer.alloc(100);
    this.cursor = 0;
  }

  /**
   * step1: Write the encoding flag to the first byte.
   * step2: Write the byte length of the string to the next four bytes.
   * step3: Write the string to the buffer.
   * @param {string} input
   */
  writeString(input) {
    if (isStringOneByteRepresentation(input)) {
      this.buffer.writeUint8(Encoding.latin1, this.cursor++);
      this.buffer.writeUint32LE(input.length, this.cursor);
      this.cursor += 4;
      this.buffer.write(input, this.cursor, 'latin1');
      this.cursor += input.length;
    } else {
      this.buffer.writeUint8(Encoding.utf16le, this.cursor++);
      this.buffer.writeUint32LE(input.length * 2, this.cursor);
      this.cursor += 4;
      this.buffer.write(input, this.cursor, 'utf16le');
      this.cursor += input.length * 2;
    }
  }

  finish() {
    return this.buffer.subarray(0, this.cursor);
  }
}

// Write strings to the buffer.
const serializer = new Serializer();
serializer.writeString('hello');
serializer.writeString('你好');
const data = serializer.finish();

// Read strings from the buffer.
const deserializer = new Deserializer(data);

> assert(deserializer.readString() === 'hello');
> assert(deserializer.readString() === '你好');

Reviewer: ```suggestion
strictEqual(deserializer.readString(), 'hello');
strictEqual(deserializer.readString(), '你好');
```

}, 1000);
```

## `v8.isOneByteRepresentation(content)`

<!-- YAML
added: REPLACEME
-->

* `content` {string}
* Returns: {boolean}

> Returns `true` if `content` uses one byte as the underlying representation Otherwise, returns `false`.

Reviewer 2: This should explain that:
- The underlying representation is ISO-8859-1 specifically
- That the underlying representation isn't just a single byte (which is what this currently says, but which obviously wouldn't be very useful)
- That this method can return `false` even for strings that are entirely representable in ISO-8859-1

}, 1000);
```

## `v8.isStringOneByteRepresentation(content)`

<!-- YAML
added: REPLACEME
-->

* `content` {string}
* Returns: {boolean}

V8 only supports `Latin-1/ISO-8859-1` and `UTF16` as the underlying representations.
If the `content` uses `Latin-1/ISO-8859-1` as the underlying representation, this function will return true;
otherwise, it returns false.

> This method returns false does not mean that the string contains some characters not in `Latin-1/ISO-8859-1`.

Reviewer 2: ```suggestion
If this method returns false, that does not mean that the string contains some characters not in `Latin-1/ISO-8859-1`.
```

Sometimes a `Latin-1` string may also be represented as `UTF16`.

```js
const { isStringOneByteRepresentation } = require('node:v8');
const bf = Buffer.alloc(20);

function writeString(input) {
  if (isStringOneByteRepresentation(input)) {
    bf.write(input, 'latin1');
  } else {
    bf.write(input, 'utf16le');

>   }

Reviewer 2: This isn't an example that would ever show up in real-world code, since there would be no way to look at the buffer and know what encoding was used to create its contents -- what is this telling users?

Author: Thank you for pointing that out. I've learned a lot about how to write good documentation.
